news/feature-4423.md

```diff
@@ -0,0 +1,60 @@
+    - default value: **no**, which means syslog-ng will attempt to continue feeding from the last remembered log position after a (re-)start
+- max-bookmark-distance
+    - integer value, maximum distance in seconds that far an earlier bookmark can point backward, e.g. if syslog-ng was stopped for 10 minutes and max-bookmark-distance is set to 60 then syslog-ng will start feeding the logs only from the last 60 seconds at startup, 9 minutes of logs 'will be lost'
+    - default value: 0, which means no limit
+- read-old-records
+    - boolean value, controls if syslog-ng should start reading logs from the oldest available at first start (or if no bookmark can be found)
+    - default value: yes
+- fetch-delay
+    - integer value, controls how much time syslog-ng should wait between reading/sending log messages, this is a fraction of a second, where wait_time = 1 second / n,  so, e.g. n=1 means that only about 1 log will be read and sent in each second, and n=1 000 000 means only 1 microsecond (the allowed minimum value now!) will be the delay between read/write attempts
+    (Use with care, though lower delay time can increase log feed performance, at the same time could lead to a heavy system load!)
+    - default value: 10 000
+- fetch-retry-delay
+    - integer value, controls how many seconds syslog-ng will wait before a repeated attempt to read/send once it's out of available logs
+    - default value: 1
+- log-fetch-limit
+    - **Warning**: _This option is now disabled due to an OSLog API bug (https://openradar.appspot.com/radar?id=5597032077066240), once it's fixed it will be enabled again_
+    - integer value, that limits the number of logs syslog-ng will send in one run
+    - default value: 0, which means no limit
+
+NOTE: the persistent OSLog store is not infinite, depending on your system setting usually, it keeps about 7 days of logs on disk, so it could happen that the above options cannot operate the way you expect, e.g. if syslog-ng was stopped for about more then a week it could happen that will not be able to restart from the last saved bookmark position (as that might not be presented in the persistent log anymore)
+
+### darwin-oslog-stream
+
+This is a wrapper around the OS command line "log stream" command that can provide a live log stream feed.
+Unlike in the case of `darwin-oslog()` the live stream can contain non-persistent log events too, so take care, there might be a huge number of log events every second that could put an unusual load on the device running syslog-ng with this source.
+Unfortunately, there's no public API to get the same programmatically, so this one is implemented using a program() source.
+
+For possible parameters please see, `log --help stream` for full reference, and `man log` for more details.
```

Comment:
The parameter type currently looks confusing can you do it like:
```suggestion
Possible parameters:
- `params`: consists of the default parameters for the `log stream` command. The default value is: `--type log --type trace --level info --level debug`
- `user-params`: consists of additional parameters for the `log stream` command if the `params` parameters are needed.

For example the following config:
`darwin-oslog-stream(params("--level info") user-params("--type log"))`
Will result in a call of `log stream --level info --type log`

*Note*: that the `--style` parameter should not be used as it will break the parsing of the messages
```

I am not sure but please check it if the `--timeout` `--source` options and the `--color` can break the parsing, as these might break the format that syslog-ng is parsing, and add them to the note part of the

Reply:
i messed up the again the param descriptions, i will fix them

--style cannot be overriden at all now as it is the last parameter so it will be used by the log tool, i guess that prt is fine as it is

Reply:
removed the user_params (i did not like it anyway), i hope this one is fine now